src/lib/appManagers/appStateManager.ts

```diff
@@ -147,156 +141,188 @@ export const STATE_INIT: State = {
       sound: false
     }
   },
-  keepSigned: true,
-  drafts: {}
+  keepSigned: true
 };
 
 const ALL_KEYS = Object.keys(STATE_INIT) as any as Array<keyof State>;
 
 const REFRESH_KEYS = ['dialogs', 'allDialogsLoaded', 'messages', 'contactsList', 'stateCreatedTime',
-  'updates', 'maxSeenMsgId', 'filters', 'topPeers'] as any as Array<keyof State>;
+  'updates', 'maxSeenMsgId', 'filters', 'topPeers', 'pinnedOrders'] as any as Array<keyof State>;
 
 export class AppStateManager extends EventListenerBase<{
   save: (state: State) => Promise<void>,
   peerNeeded: (peerId: number) => void,
   peerUnneeded: (peerId: number) => void,
 }> {
   public static STATE_INIT = STATE_INIT;
-  public loaded: Promise<State>;
+  private loaded: Promise<State>;
   private log = logger('STATE'/* , LogLevels.error */);
 
   private state: State;
-  private savePromise: Promise<void>;
-  private tempId = 0;
 
   private neededPeers: Map<number, Set<string>> = new Map();
   private singlePeerMap: Map<string, number> = new Map();
 
+  public storages = {
+    users: new AppStorage<Record<number, User>>({
+      storeName: 'users'
+    }),
+
+    chats: new AppStorage<Record<number, Chat>>({
+      storeName: 'chats'
+    }),
+
+    dialogs: new AppStorage<Record<number, Dialog>>({
+      storeName: 'dialogs'
+    })
+  };
+
+  public storagesResults: {[key in keyof AppStateManager['storages']]: any[]} = {} as any;
+
   constructor() {
     super();
     this.loadSavedState();
   }
 
-  public loadSavedState() {
+  public loadSavedState(): Promise<State> {
     if(this.loaded) return this.loaded;
-    //console.time('load state');
-    return this.loaded = new Promise((resolve) => {
-      Promise.all(ALL_KEYS.concat('user_auth' as any).map(key => sessionStorage.get(key))).then((arr) => {
-        let state: State = {};
+    console.time('load state');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MikeOwino/tweb/pullRequest?prid=7329411)
```suggestion
    console.time("load state");
```